src/OrbitBase/include/OrbitBase/JoinFutures.h

```diff
@@ -6,13 +6,88 @@
 #define ORBIT_BASE_JOIN_FUTURES_H_
 
 #include "OrbitBase/Future.h"
+#include "OrbitBase/FutureHelpers.h"
+#include "OrbitBase/Promise.h"
 #include "absl/types/span.h"
 
+namespace orbit_base_internal {
+
+template <typename T>
+class SharedStateJoin {
+  orbit_base::Promise<std::vector<T>> promise;
+  absl::Mutex mutex;
+  int incompleted_futures;
```

Comment:
incomplete_futures

Reply:
Also, it seems like this can be size_t, can it be negative?

Reply:
Yes and yes, it can be size_t. Fixed. Thanks!